cfme/tests/automate/custom_button/test_infra_objects.py

```diff
@@ -193,3 +196,63 @@ def test_custom_button_automate(appliance, request, submit, setup_obj, button_gr
             assert False, "Expected {} requests not found in automation log".format(
                 str(expected_count)
             )
+
+
+@pytest.mark.meta(
+    blockers=[
+        BZ(
+            1635797,
+            forced_streams=["5.9", "5.10"],
+            unblock=lambda button_group: bool(
+                [obj for obj in ["HOST", "VM_INSTANCE", "TEMPLATE_IMAGE"] if obj in button_group]
+            ),
+        )
+    ]
+)
+def test_custom_button_dialog(appliance, dialog, request, setup_obj, button_group):
+    """ Test custom button with dialog and InspectMe method
+
+    Prerequisites:
+        * Appliance with Infra provider
+        * Simple TextInput service dialog
+
+    Steps:
+        * Create custom button group with the Object type
+        * Create a custom button with service dialog
+        * Navigate to object Details page
+        * Check for button group and button
+        * Select/execute button from group dropdown for selected entities
+        * Fill dialog and submit
+        * Check for the proper flash message related to button execution
+
+    Bugzillas:
+        * 1635797, 1555331, 1574403, 1640592
+    """
+
+    group, obj_type = button_group
+
+    # Note: No need to set display_for dialog only work with Single entity
+    button = group.buttons.create(
+        text=fauxfactory.gen_alphanumeric(),
+        hover=fauxfactory.gen_alphanumeric(),
+        dialog=dialog,
+        system="Request",
+        request="InspectMe",
+    )
+    request.addfinalizer(button.delete_if_exists)
+
+    view = navigate_to(setup_obj, "Details")
+    custom_button_group = Dropdown(view, group.hover)
+    assert custom_button_group.has_item(button.text)
+    custom_button_group.item_select(button.text)
+
+    dialog_view = view.browser.create_view(TextInputDialogView)
+    dialog_view.wait_displayed()
+    assert dialog_view.service_name.fill("Custom Button Execute")
+    dialog_view.submit.click()
+
+    if not (
+        BZ(bug_id=1640592, forced_streams=["5.9", "5.10"]).blocks and obj_type == "TEMPLATE_IMAGE"
+    ):
+        view.wait_displayed("60s")
+    view.flash.assert_message("Order Request was Submitted")
```

Comment:
Question : Can you also check if the order request completed successfully for both cloud and infra objects ?

Reply:
@sshveta agreed with you... but custom button simple dialog submission did not create any request. So for better testing, I am adding `request` checks from `automation log`